Makefile

```diff
@@ -15,205 +15,182 @@
+	DEFINES += APP_FEATURE_POOL_RETIREMENT
+	DEFINES += APP_FEATURE_BYRON_ADDRESS_DERIVATION
+	DEFINES += APP_FEATURE_BYRON_PROTOCOL_MAGIC_CHECK
 endif
+# always include this, it's important for Plutus users
+DEFINES += APP_FEATURE_TOKEN_MINTING
 
-################
-# Default rule #
-################
-
-all: default
-
-
-##############
-#   Build    #
-##############
-
-# import generic rules from the sdk
-include $(BOLOS_SDK)/Makefile.rules
-
-#add dependency on custom makefile filename
-dep/%.d: %.c Makefile
-
-listvariants:
-	@echo VARIANTS COIN cardano_ada
-
-# part of CI
-analyze: clean
-	scan-build --use-cc=clang -analyze-headers -enable-checker security -enable-checker unix -enable-checker valist -o scan-build --status-bugs make default
-
-##############
-#   Load     #
-##############
+########################################
+#     Application custom permissions   #
+########################################
+# See SDK `include/appflags.h` for the purpose of each permission
+#HAVE_APPLICATION_FLAG_DERIVE_MASTER = 1
+#HAVE_APPLICATION_FLAG_GLOBAL_PIN = 1
+HAVE_APPLICATION_FLAG_BOLOS_SETTINGS = 1
+#HAVE_APPLICATION_FLAG_LIBRARY = 1
+
+########################################
+# Application communication interfaces #
+########################################
+ENABLE_BLUETOOTH = 1
+#ENABLE_NFC = 1
+
+########################################
+#         NBGL custom features         #
+########################################
+ENABLE_NBGL_QRCODE = 1
+#ENABLE_NBGL_KEYBOARD = 1
+#ENABLE_NBGL_KEYPAD = 1
+
+########################################
+#          Features disablers          #
+########################################
+# These advanced settings allow to disable some feature that are by
+# default enabled in the SDK `Makefile.standard_app`.
+#DISABLE_STANDARD_APP_FILES = 1
+#DISABLE_DEFAULT_IO_SEPROXY_BUFFER_SIZE = 1 # To allow custom size declaration
+#DISABLE_STANDARD_APP_DEFINES = 1 # Will set all the following disablers
+#DISABLE_STANDARD_SNPRINTF = 1
+#DISABLE_STANDARD_USB = 1
+#DISABLE_STANDARD_WEBUSB = 1
+#DISABLE_STANDARD_BAGL_UX_FLOW = 1
+
+########################################
+#       Additional configuration       #
+########################################
+
+# USB U2F
+DEFINES += HAVE_U2F HAVE_IO_U2F U2F_PROXY_MAGIC=\"ADA\"
+SDK_SOURCE_PATH  += lib_u2f
+
+DEFINES += HAVE_UX_LEGACY
+
+# Protect against stack overflows
+DEFINES += HAVE_BOLOS_APP_STACK_CANARY
 
-NANOS_ID = 1
+# mnemonic and PIN for testing on a physical device
 WORDS = "abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about"
 PIN = 5555
 
-APP_LOAD_PARAMS =--appFlags 0x200 --curve ed25519 --path "44'/1815'" --path "1852'/1815'" --path "1853'/1815'" --path "1854'/1815'" --path "1855'/1815'" --path "1694'/1815'"
-APP_LOAD_PARAMS += $(COMMON_LOAD_PARAMS)
+# based in part on https://interrupt.memfault.com/blog/best-and-worst-gcc-clang-compiler-flags
+WERROR   := -Werror=return-type -Werror=parentheses -Werror=format-security
```

Comment:
`-Werror=format-security` this warning is already enabled in the SDK.
Regarding the whole line, I would recommend adding theses as warning, which are converted as errors only for CI builds.